vendor/github.com/raintank/schema/metric.go

```diff
@@ -251,6 +255,26 @@ func MetricDefinitionFromMetricData(d *MetricData) *MetricDefinition {
 	return md
 }
 
+// SanitizeNameAsTagValue takes a name and potentially
+// modifies it to ensure it is a valid value that can be
+// used as a tag value. This is important when we index
+// metric names as values of the tag "name"
+func SanitizeNameAsTagValue(name string) string {
+	if !strings.Contains(name, "~") {
```

Comment:
Why all the complexity in this function. Cant you just use `strings.ReplaceAll(name, "~", "")`?

Reply:
I definitely can, but since this is a function that may get called very often (potentially millions of times in one query) I wanted to keep it as efficient as possible

Reply:
Actually, the definition of what is valid has just changed, so this function will get updated anyway

Reply:
I pushed a new version. I benchmarked it and compared it with an alternative version which is simpler by relying on `strings`:

```
func SanitizeNameAsTagValue(name string) (string, error) {
	if len(name) == 0 {
		return name, fmt.Errorf("Invalid name of length 0")
	}

	if name[0] != '~' {
		return name, nil
	}

	name = strings.TrimLeft(name, "~")

	if len(name) == 0 {
		return "", fmt.Errorf("Invalid name, requiring other characters than ~")
	}

	return name, nil
}
```

The first one is with the latest commit that I pushed (f065be1c4c09096e3c0a787fe966831d8e097fe1) and the second is with the above simplified version:
```
replay@vbox:~/go/src/github.com/raintank/schema$ go test -run=none -bench=BenchmarkNameSanitized
goos: linux
goarch: amd64
pkg: github.com/raintank/schema
BenchmarkNameSanitizedAsTagValueWithValidValue-3        300000000                5.02 ns/op
BenchmarkNameSanitizedAsTagValueWithInvalidValue-3      200000000                6.10 ns/op
PASS
ok      github.com/raintank/schema      3.867s
replay@vbox:~/go/src/github.com/raintank/schema$ go test -run=none -bench=BenchmarkNameSanitized
goos: linux
goarch: amd64
pkg: github.com/raintank/schema
BenchmarkNameSanitizedAsTagValueWithValidValue-3        300000000                5.09 ns/op
BenchmarkNameSanitizedAsTagValueWithInvalidValue-3      20000000                84.8 ns/op
PASS
ok      github.com/raintank/schema      3.810s
```

Reply:
Actually, I don't think it makes sense for this method to have the potential to return errors, if the returned value is `""` then the caller should just have to deal with it, i'll change that

Reply:
I changed it again so the method does not return errors and compared the latest version (https://github.com/raintank/schema/commit/1adcebf5a5f44be63dc9df90e72dae7696b47db2) with a simpler version relying on `strings`. it's quite a difference:

```
replay@vbox:~/go/src/github.com/raintank/schema$ go test -run=none -bench=BenchmarkNameSanitized
goos: linux
goarch: amd64
pkg: github.com/raintank/schema
BenchmarkNameSanitizedAsTagValueWithValidValue-3        300000000                4.41 ns/op
BenchmarkNameSanitizedAsTagValueWithInvalidValue-3      300000000                5.74 ns/op
PASS
ok      github.com/raintank/schema      4.095s
replay@vbox:~/go/src/github.com/raintank/schema$ go test -run=none -bench=BenchmarkNameSanitized
goos: linux
goarch: amd64
pkg: github.com/raintank/schema
BenchmarkNameSanitizedAsTagValueWithValidValue-3        20000000                70.4 ns/op
BenchmarkNameSanitizedAsTagValueWithInvalidValue-3      20000000                82.0 ns/op
PASS
ok      github.com/raintank/schema      3.220s
```